plugins/node/opentelemetry-instrumentation-aws-lambda/src/instrumentation.ts

```diff
@@ -97,14 +97,28 @@ export class AwsLambdaInstrumentation extends InstrumentationBase<AwsLambdaInstr
     // Lambda loads user function using an absolute path.
     let filename = path.resolve(taskRoot, moduleRoot, module);
     if (!filename.endsWith('.js')) {
-      // its impossible to know in advance if the user has a cjs or js file.
-      // check that the .js file exists otherwise fallback to next known possibility
+      // Its impossible to know in advance if the user has a js, cjs or mjs file.
+      // Check that the .js file exists otherwise fallback to the next known possibilities (.cjs, .mjs).
       try {
         fs.statSync(`${filename}.js`);
         filename += '.js';
       } catch (e) {
-        // fallback to .cjs
-        filename += '.cjs';
+        try {
+          fs.statSync(`${filename}.cjs`);
+          // fallback to .cjs (CommonJS)
+          filename += '.cjs';
+        } catch (e2) {
+          try {
+            fs.statSync(`${filename}.mjs`);
+            // fallback to .mjs (ESM)
+            filename += '.mjs';
+          } catch (e3) {
+            diag.error(
```

Comment:
Or this could be a diag.warn. I'm not sure if we have clear/clean policy on when to use which level.


Suggest use the instrunmentation's scoped `this._diag` so the insrtumentation name is included in the diag message.

```suggestion
            this._diag.error(
```

Reply:
done